lib/plugins/aws/package/compile/events/apiGateway/index.js

```diff
@@ -264,6 +264,33 @@ class AwsCompileApigEvents {
           );
         }
 
+        if (
+          this.serverless.service.provider.name === 'aws' &&
+          Object.values(this.serverless.service.functions).some(({ events }) =>
+            events.some(({ http }) => {
+              if (http && http.authorizer) {
+                if (_.isObject(http.authorizer)) {
+                  // Deprecation should not apply to already existing, external authorizers
+                  if (http.authorizer.authorizerId) return false;
+                  // Deprecation should not apply to `aws_iam` authorization type
+                  if (http.authorizer.type && http.authorizer.type.toUpperCase() === 'AWS_IAM') {
+                    return false;
+                  }
+                  return http.authorizer.identitySource === undefined;
+                }
+                // Authorizer is defined as a string and we want to check if its `aws_iam` - in such case the deprecation will not apply
+                return !(http.authorizer.toUpperCase() === 'AWS_IAM');
+              }
+              return false;
+            })
+          )
+        ) {
+          this.serverless._logDeprecation(
+            'AWS_API_GATEWAY_DEFAULT_IDENTITY_SOURCE',
+            'Starting with `v3.0.0`, "functions[].events[].http.authorizer.identitySource" will no longer be set to "method.request.header.Authorization" by default. If you want to keep this setting, please set it explicitly in your configuration. If you do not want this to be set, please set it explicitly to `null`.'
```

Comment:
In CLI logs we tend to not use back-ticks, we eventually wrap with regular quotes. Here I think there's no need to quote `v3.0.0`, and I would wrap `null` with quotes.

Let's also maybe add new line after first sentence.